pkg/kv/kvserver/store_raft.go

```diff
@@ -320,6 +322,15 @@ func (s *Store) HandleRaftUncoalescedRequest(
 	return enqueue
 }
 
+// HandleRaftRequestSent is called to capture outgoing Raft messages just prior
+// to their transmission to the raftSendQueue. Note that the message might not
+// be successfully queued if it gets dropped by SendAsync due to a full outgoing
+// queue.
+func (s *Store) HandleRaftRequestSent(ctx context.Context, req *kvserverpb.RaftMessageRequest) {
+	comparisonResult := s.getLocalityComparison(ctx, req.FromReplica.NodeID, req.ToReplica.NodeID)
+	s.metrics.updateRaftMetricOnOutgoingMsg(comparisonResult, int64(req.Size()))
```

Comment:
Could you pass the size in instead? We don't want to compute it twice, this is a hot path. I believe the caller computes the size right around the time of the call into this method.

Reply:
Done. Changed `HandleRaftRequestSent` to takes in specific request properties instead and made a note on the comment.